helix-core/src/main/java/org/apache/helix/model/InstanceConfig.java

```diff
@@ -505,6 +505,46 @@ public void setMaxConcurrentTask(int maxConcurrentTask) {
     _record.setIntField(InstanceConfigProperty.MAX_CONCURRENT_TASK.name(), maxConcurrentTask);
   }
 
+  /**
+   * Get the instance capacity information from the map fields
+   * @return data map if it exists, or empty map
+   */
+  public Map<String, Integer> getInstanceCapacityMap() {
+    Map<String, String> capacityData = _record.getMapField(InstanceConfigProperty.INSTANCE_CAPACITY_MAP.name());
+    Map<String, Integer> capacityDataMap = new HashMap<>();
+
+    if (capacityData != null && !capacityData.isEmpty()) {
```

Comment:
Really suggest Java8 style, less error-prone and easier to read. Same applies to the rest of conversions here and there

Reply:
Tried, only reduce one line... About error-prone, and easy read, I don't see a big difference actually.
We are reading and manipulating 2 maps here, Lambda style is more confusing from my perspective.

Reply:
```
capacityDataMap.entrySet().stream().collect(Collectors.toMap(e -> e.getKey(), e -> Integer.parseInt(e.getValue())))
```
or better with Guava's library
```
Maps.transformValues(capacityDataMap, Integer::parseInt)
```
Why is this error prone?  You don't need to check if map is empty or not

Reply:
If it's a for loop, I don't need to check for empty as well, right?